In [76]:
import pandas as pd # type: ignore
import numpy as np # type: ignore
pd.options.display.float_format = '{:.0f}'.format


df = pd.read_csv('prodgy_population_task1.csv' , skiprows=5)
world_population_all_years = df[df['Country Name'] == 'World']
df.head()

Country Name Country Code     Indicator Name Indicator Code  \
0                        Aruba          ABW  Population, total    SP.POP.TOTL   
1  Africa Eastern and Southern          AFE  Population, total    SP.POP.TOTL   
2                  Afghanistan          AFG  Population, total    SP.POP.TOTL   
3   Africa Western and Central          AFW  Population, total    SP.POP.TOTL   
4                       Angola          AGO  Population, total    SP.POP.TOTL   

       1960      1961      1962      1963      1964      1965  ...      2015  \
0     54608     55811     56682     57475     58178     58782  ...    104257   
1 130692579 134169237 137835590 141630546 145605995 149742351  ... 600008424   
2   8622466   8790140   8969047   9157465   9355514   9565147  ...  33753499   
3  97256290  99314028 101445032 103667517 105959979 108336203  ... 408690375   
4   5357195   5441333   5521400   5599827   5673199   5736582  ...  28127721   

       2016      2017      2018      2019      2020      2021      2022  2023  \
0    104874    105439    105962    106442    106585    106537    106445   NaN   
1 616377605 632746570 649757148 667242986 685112979 702977106 720859132   NaN   
2  34636207  35643418  36686784  37769499  38972230  40099462  41128771   NaN   
3 419778384 431138704 442646825 454306063 466189102 478185907 490330870   NaN   
4  29154746  30208628  31273533  32353588  33428486  34503774  35588987   NaN   

   Unnamed: 68  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN  

[5 rows x 69 columns]

In [67]:
world_population_all_years

Country Name Country Code     Indicator Name Indicator Code       1960  \
259        World          WLD  Population, total    SP.POP.TOTL 3031474234   

          1961       1962       1963       1964       1965  ...       2015  \
259 3072421801 3126849612 3193428894 3260441925 3328209022  ... 7403850164   

          2016       2017       2018       2019       2020       2021  \
259 7490415449 7576441961 7660371127 7741774583 7820205606 7888305693   

          2022  2023  Unnamed: 68  
259 7950946801   NaN          NaN  

[1 rows x 69 columns]

READING CSV:

Turns out we can not read a csv file if there is irregular pattern somewhere just as we have in the start. So to fix that we use this "skiprows" parameter of pandas read_csv function.

CLEANING DATA:

This dataset had :
- 68 unnamed columns 
- one empty column named: 2023

In [ ]:
columns_to_remove = ['2023']
df =df.drop(columns=columns_to_remove)


In [37]:

df = df.dropna(axis=1, how='all')
df.head()

Country Name Country Code     Indicator Name Indicator Code  \
0                        Aruba          ABW  Population, total    SP.POP.TOTL   
1  Africa Eastern and Southern          AFE  Population, total    SP.POP.TOTL   
2                  Afghanistan          AFG  Population, total    SP.POP.TOTL   
3   Africa Western and Central          AFW  Population, total    SP.POP.TOTL   
4                       Angola          AGO  Population, total    SP.POP.TOTL   

          1960         1961         1962         1963         1964  \
0      54608.0      55811.0      56682.0      57475.0      58178.0   
1  130692579.0  134169237.0  137835590.0  141630546.0  145605995.0   
2    8622466.0    8790140.0    8969047.0    9157465.0    9355514.0   
3   97256290.0   99314028.0  101445032.0  103667517.0  105959979.0   
4    5357195.0    5441333.0    5521400.0    5599827.0    5673199.0   

          1965  ...         2013         2014         2015         2016  \
0      58782.0  ...     102880.0     103594.0     104257.0     104874.0   
1  149742351.0  ...  567892149.0  583651101.0  600008424.0  616377605.0   
2    9565147.0  ...   31541209.0   32716210.0   33753499.0   34636207.0   
3  108336203.0  ...  387204553.0  397855507.0  408690375.0  419778384.0   
4    5736582.0  ...   26147002.0   27128337.0   28127721.0   29154746.0   

          2017         2018         2019         2020         2021  \
0     105439.0     105962.0     106442.0     106585.0     106537.0   
1  632746570.0  649757148.0  667242986.0  685112979.0  702977106.0   
2   35643418.0   36686784.0   37769499.0   38972230.0   40099462.0   
3  431138704.0  442646825.0  454306063.0  466189102.0  478185907.0   
4   30208628.0   31273533.0   32353588.0   33428486.0   34503774.0   

          2022  
0     106445.0  
1  720859132.0  
2   41128771.0  
3  490330870.0  
4   35588987.0  

[5 rows x 67 columns]

REMOVED!


In [40]:
print(len(df['Country Name']))

266


There are about 266 countries mentioned in this data set
now, lets find out the country with largest population in 1960

Largest Population in 1960:

In [48]:

countries_in_1960 = df[['Country Name','1960']]
countries_in_1960.head()

Country Name         1960
0                        Aruba      54608.0
1  Africa Eastern and Southern  130692579.0
2                  Afghanistan    8622466.0
3   Africa Western and Central   97256290.0
4                       Angola    5357195.0

In [60]:
largest_population = df.nlargest(5,'1960')[['Country Name','1960']]
largest_population.head()


Country Name       1960
259                World 3031474234
103     IDA & IBRD total 2297972283
140  Low & middle income 2243178704
156        Middle income 2107416130
102            IBRD only 1904346683

COUNTRY "WORLD" :
When searching for the country with the greatest population in 1960, I encountered an error in the csv. It appears that there is a row labeled "World" in the dataset. This row should not be included in the analysis as it represents a global population aggregate rather than an individual country.

to deal with it ,  we are going to ignore it when comparing so lets first save it in a df(on top "world_population_all_years")

In [87]:
filtered_df = df[df['Country Name'] != 'World']
filtered_df = filtered_df[~filtered_df['Country Name'].str.contains('income|only|&|dividend|members' , case=False)]
largest_population = filtered_df.nlargest(5,'1960')[['Country Name','1960']]
largest_population.head()

Country Name      1960
40            China 667070000
204      South Asia 571192428
109           India 445954579
104       IDA total 393625600
73   European Union 356947129